### DISCUSSION TITLE
numpy min error

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [python/typing#1126] [Q&A]
```
def min_vector(v: npt.ArrayLike) -> np.ndarray:
    return (v - v.min(0))
```
The function takes a numpy vector and performs a numpy min operation on it but mypy generates the error:

`error: Item "_SupportsArray[dtype[Any]]" of "Union[_SupportsArray[dtype[Any]], _NestedSequence[_SupportsArray[dtype[Any]]], bool, int, float, complex, str, bytes, _NestedSequence[Union[bool, int, float, complex, str, bytes]]]" has no attribute "min"`

**Comment by BvB93** [comment]
This is more of a numpy-specific question, but the reason why this doesn't work is that [`ArrayLike`](https://numpy.org/doc/stable/reference/typing.html?highlight=typing#numpy.typing.ArrayLike) represents objects that can be _coerced_ into an `ndarray`, not ones that _duck-type_ as an `ndarray`. Examples of this would include things like integers or lists of floats, both of which lack a `min` method.

**Reply by dineshbvadhia** [reply]
Is the question for numpy developers or is there an answer here? Thanks
